src/test/e2e/00_use_cli_test.go

```diff
@@ -16,126 +16,174 @@ import (
-
-	pkgName := fmt.Sprintf("zarf-package-dos-games-%s.tar.zst", e2e.Arch)
-
-	_ = os.Mkdir(otherTmpPath, 0750)
-	stdOut, stdErr, err = e2e.ExecZarfCommand("package", "create", "examples/dos-games", "--confirm", "--zarf-cache", cachePath, "--tmpdir", otherTmpPath, "--log-level=debug")
-	require.Contains(t, stdErr, otherTmpPath, "The other tmp path should show as being created")
-	require.NoError(t, err, stdOut, stdErr)
-
-	stdOut, stdErr, err = e2e.ExecZarfCommand("package", "inspect", pkgName, "--tmpdir", otherTmpPath, "--log-level=debug")
-	require.Contains(t, stdErr, otherTmpPath, "The other tmp path should show as being created")
-	require.NoError(t, err, stdOut, stdErr)
-
-	e2e.CleanFiles(pkgName)
-
-	files, err := os.ReadDir(imageCachePath)
-	require.NoError(t, err, "Encountered an unexpected error when reading image cache path")
-	require.Greater(t, len(files), 1)
-
-	// Test removal of cache
-	stdOut, stdErr, err = e2e.ExecZarfCommand("tools", "clear-cache", "--zarf-cache", cachePath)
-	require.NoError(t, err, stdOut, stdErr)
-
-	// Check that ReadDir returns no such file or directory for the cachePath
-	_, err = os.ReadDir(cachePath)
-	if runtime.GOOS == "windows" {
-		msg := fmt.Sprintf("open %s: The system cannot find the file specified.", cachePath)
-		require.EqualError(t, err, msg, "Did not receive expected error when reading a directory that should not exist")
-	} else {
-		msg := fmt.Sprintf("open %s: no such file or directory", cachePath)
-		require.EqualError(t, err, msg, "Did not receive expected error when reading a directory that should not exist")
-	}
-
-	// Test generation of PKI
-	tlsCA := "tls.ca"
-	tlsCert := "tls.crt"
-	tlsKey := "tls.key"
-	stdOut, stdErr, err = e2e.ExecZarfCommand("tools", "gen-pki", "github.com", "--sub-alt-name", "google.com")
-	require.NoError(t, err, stdOut, stdErr)
-	require.Contains(t, stdErr, "Successfully created a chain of trust for github.com")
-
-	_, err = os.ReadFile(tlsCA)
-	require.NoError(t, err)
-
-	_, err = os.ReadFile(tlsCert)
-	require.NoError(t, err)
-
-	_, err = os.ReadFile(tlsKey)
-	require.NoError(t, err)
-
-	e2e.CleanFiles(shasumTestFilePath, cachePath, otherTmpPath, pkgName, tlsCA, tlsCert, tlsKey)
+
+	t.Run("zarf prepare sha256sum <local>", func(t *testing.T) {
+		t.Parallel()
+
+		// Test `zarf prepare sha256sum` for a local asset
+		expectedShasum := "61b50898f982d015ed87093ba822de0fe011cec6dd67db39f99d8c56391a6109\n"
+		shasumTestFilePath := "shasum-test-file"
+
+		// run `zarf package create` with a specified tmp location
+		otherTmpPath := t.TempDir()
+
+		e2e.CleanFiles(shasumTestFilePath, otherTmpPath)
+
+		err := os.WriteFile(shasumTestFilePath, []byte("random test data 🦄\n"), 0600)
+		require.NoError(t, err)
+
+		stdOut, stdErr, err := e2e.Zarf("prepare", "sha256sum", shasumTestFilePath)
+		require.NoError(t, err, stdOut, stdErr)
+		require.Equal(t, expectedShasum, stdOut, "The expected SHASUM should equal the actual SHASUM")
+		t.Cleanup(func() {
```

Comment:
Here and elsewhere;

`t.Cleanup( ... )` should be called BEFORE any `require.()` checks so they don't remain during failed tests.